### PR TITLE
Client remake

### PR DESCRIPTION
**Pull request by unbrokenguy**
New client architecture, which makes the client more flexible to add new functionality

### CODE REVIEW

File: client/states.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            file_name_input = input()
+            if files[str(file_name_input)] in base_actions.keys():
+                return base_actions[files[file_name_input]]
+            if file_name_input not in files.keys():
+                Tools.print_error("Пожалуйста введите корректные данные.")
+            else:
+                self.download(categories[category_input], files[file_name_input])
+        return None
+
+    def action(self):
```
Comment (ARManakhov) [inline client/states.py:115]: recommend you to split action() into separate functions to reduce the complexity

**Review by ARManakhov — CHANGES_REQUESTED** [review]

File: client/states.py (hunk 1 above)
Comment (unbrokenguy) [reply client/states.py:115]: So you say that DownloadState.action() is too big and I need to split it to separate functions?

Comment (ARManakhov) [reply client/states.py:115]: yes 😋

Comment (unbrokenguy) [reply client/states.py:115]: From my point of view, DownloadState.action () has normal size, perhaps you want to point out that DownloadState.handle_download () in client.py was 2 methods, the first for selecting a category and the second for downloading data?

**Pull request closed without merging**